src/vm/profilinghelper.cpp

```diff
@@ -1193,9 +1193,14 @@ HRESULT ProfilingAPIUtility::LoadProfiler(
         // and kill objects without relocating and thus not doing a heap walk.
         if (CORProfilerTrackGC())
         {
-            LOG((LF_CORPROF, LL_INFO10, "**PROF: Turning off concurrent GC at startup.\n"));       
-            g_pConfig->SetGCconcurrent(0);
-            LOG((LF_CORPROF, LL_INFO10, "**PROF: Concurrent GC has been turned off at startup.\n"));       
+            LOG((LF_CORPROF, LL_INFO10, "**PROF: Turning off concurrent GC at startup.\n"));
+            // Previously we would use SetGCConcurrent(0) to indicate to the GC that it shouldn't even
+            // attempt to use concurrent GC. The standalone GC feature create a cycle during startup,
+            // where the profiler couldn't set startup flags for the GC. To overcome this, we call
+            // TempraryDisableConcurrentGC and never enable it again. This has a slight perf cost,
+            // since the GC will create concurrent GC data structures, but it shouldn't be measurable.
```

Comment:
I would not say this. The memory we use to track for concurrent GC certainly can be measurable. But I am ok with doing this - I don't think profiling code in general enforces high perf requirements.

Reply:
David I was just chatting with Maoni. On 32 bit that datastructure is 1/64 the size of the entire GC heap so its easily measurable. I suggest we update the comment to something like this: "This has some perf cost since the GC will create concurrent GC data structures, but we think it is acceptable in the context of this kind of profiling" Could you also update the commit text to reference the bug you were fixing. Otherwise, LGTM - Thanks!

Reply:
Thanks, I updated the comment to similar wording to what you suggested